stdlib/immutablepriorityqueue.md

```diff
@@ -0,0 +1,248 @@
+---
+title: ImmutablePriorityQueue
+---
+
+An immutable priority queue. A priority queue is a data structure that maintains elements in a priority order. Elements with higher priority are served before elements with lower priority when extracting from the priority queue.
+
+<details disabled>
+<summary tabindex="-1">Added in <code>next</code></summary>
+No other changes yet.
+</details>
+
+```grain
+import ImmutablePriorityQueue from "immutablepriorityqueue"
+```
+
+## Types
+
+Type declarations included in the ImmutablePriorityQueue module.
+
+### Immutablepriorityqueue.**ImmutablePriorityQueue**
```

Comment:
@ospencer we've never had this type of module name so graindoc doesn't camelcase it. Thoughts?

Reply:
This will be fixed by #1406 so we can merge this PR and then fix up the generated docs in that PR